proposals/NNNN-refining-property-wrapper-related-initialization.md

```diff
@@ -0,0 +1,237 @@
+# Refining property-wrapper-related initialization
+
+* Proposal: [SE-NNNN](NNNN-refining-property-wrapper-related-initialization.md)
+* Authors: [Amritpan Kaur](https://github.com/amritpan), [Filip Sakel](https://github.com/filip-sakel), [Frederick Kellison-Linn](https://github.com/jumhyn)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+
+## Introduction
+
+[SE 0258](https://github.com/apple/swift-evolution/blob/master/proposals/0258-property-wrappers.md) introduced property wrappers and [SE 0293](https://github.com/apple/swift-evolution/blob/main/proposals/0293-extend-property-wrappers-to-function-and-closure-parameters.md#detailed-design) expanded them with function-like declarations. Today, property wrapper initialization exhibits inconsistencies due to its growing versatility. Specifically, memberwise initializers use complex, poorly documented rules and projection initialization remains limited. This proposal will simplify synthesized memberwise initialization for types with wrapped properties and extend projection value initialization to include global, type, and local wrapped properties.
+
+## Motivation
+
+Property wrappers were initially adopted in  [SE 0258](https://github.com/apple/swift-evolution/blob/main/proposals/0258-property-wrappers.md) and expanded to function parameters and closures in  [SE 0293](https://github.com/apple/swift-evolution/blob/main/proposals/0293-extend-property-wrappers-to-function-and-closure-parameters.md). This two-phase adoption of property wrappers throughout the language has left property wrappers considered as a whole with some inconsistencies and complexity that is no longer necessary.
+
+Today, the rules for how wrapped properties are represented in the synthesized memberwise initializer for structs are poorly documented and quite complex. To summarize the points from [this thread](https://forums.swift.org/t/does-the-new-swift-5-5-init-projectedvalue-functionality-not-work-with-synthesized-memberwise-initializers/51232/7), the rules are:
```

Comment:
@Jumhyn @amritpan Do you think these points would look better as comments in the following code examples (to avoid repetition)?

Reply:
I do yes. Maybe something like (i.e. `MyView(value: Int)`) or ```code block ``` ? A SwiftUI and a general example

Reply:
Yep moving these inline makes sense to me now that we have a concrete example attached.

Reply:
We already have a SwiftUI example in Source Compatibility, so perhaps we could include a wrapper from other frameworks. (Personally, I don't mind SwiftUI examples, but we received feedback on this in 293.)